ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -138,6 +156,24 @@ private void initCacheFromBackupFile() {
     // Refresh last modified time of Accounts and Containers in cache
     accountInfoMap.refreshLastModifiedTime();
     accountInfoMapRef.set(accountInfoMap);
+
+    if (config.enableNewDatabaseForMigration) {
+      Collection<Account> accountsNew = backupFileManagerNew.getLatestAccounts(aMonthAgo);
+      AccountInfoMap accountInfoMapNew = null;
+      if (accountsNew != null) {
+        try {
+          accountInfoMapNew = new AccountInfoMap(accountsNew);
+        } catch (Exception e) {
+          logger.warn("Failure in parsing of Account Metadata from local backup file", e);
+        }
+      }
+      if (accountInfoMapNew == null) {
+        accountInfoMapNew = new AccountInfoMap(accountServiceMetrics);
+      }
+      // Refresh last modified time of Accounts and Containers in cache
+      accountInfoMapNew.refreshLastModifiedTime();
+      accountInfoMapRefNew.set(accountInfoMapNew);
+    }
```

Comment:
We could just copy accountInfoMapNew from accountInfoMap

Reply:
There are two caches, accountInfoMapRef is for old db and the data comes from backupFileManager. accountInfoMapRefNew is for new db and data comes from backupFileManagerNew. there's a chance that those two are different during transition time. And after migration, I'm planning to remove the old logic when we tested the two database are in sync.